backend/infrahub/core/node/delete_validator.py

```diff
@@ -37,8 +38,8 @@ def _add_to_dependency_graph(
         if kind not in self._dependency_graph:
             self._dependency_graph[kind] = {}
         if relationship_type not in self._dependency_graph[kind]:
-            self._dependency_graph[kind][relationship_type] = {}
-        self._dependency_graph[kind][relationship_type][relationship_identifier] = peer_kind
+            self._dependency_graph[kind][relationship_type] = defaultdict(set)
+        self._dependency_graph[kind][relationship_type][relationship_identifier].add(peer_kind)
```

Comment:
if a mandatory relationship is defined on a generic, then there can be multiple peer kinds that use the same relationship identifier, so this data structure needs to be a set instead of a single item that can be overwritten